lib/internal/test_runner/reporter/dot.js

```diff
@@ -1,17 +1,27 @@
 'use strict';
+const { MathMax } = primordials;
 
 module.exports = async function* dot(source) {
   let count = 0;
+  let columns = getLineLength();
   for await (const { type } of source) {
     if (type === 'test:pass') {
       yield '.';
     }
     if (type === 'test:fail') {
       yield 'X';
     }
-    if ((type === 'test:fail' || type === 'test:pass') && ++count % 20 === 0) {
+    if ((type === 'test:fail' || type === 'test:pass') && ++count === columns) {
```

Comment:
Could the `++count === columns` be nested inside of the top level `if`? That way we only need to call `getLineLength()` once from there.

Reply:
The need to call again is in case of resize of the terminal